addons/package_hierarchy/models/package_links.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class PackageHierarchyLink(models.Model):
+    _name = "package.hierarchy.link"
```

Comment:
A class docstring explaining the purpose of the links would be very helpful, describing for what they are used for.